File: backend/app/api/routes/knowledge.py
from fastapi import APIRouter, Query

from app.schemas.knowledge import KnowledgeDocumentsResponse, KnowledgeSearchResponse
from app.services.knowledge_service import KnowledgeService

router = APIRouter()


@router.get("/documents", response_model=KnowledgeDocumentsResponse)
def list_documents() -> KnowledgeDocumentsResponse:
    return KnowledgeService().list_documents()


@router.get("/search", response_model=KnowledgeSearchResponse)
def search_knowledge(
    q: str = Query(min_length=1),
    limit: int = Query(default=5, ge=1, le=20),
) -> KnowledgeSearchResponse:
    return KnowledgeService().search(q, limit)

File: frontend/src/shared/api/knowledge.ts
import type { KnowledgeDocumentsResponse, KnowledgeSearchResponse } from "../types/knowledge";

const apiBaseUrl = import.meta.env.VITE_API_BASE_URL ?? "http://localhost:8000/api";

async function requestJson<T>(path: string): Promise<T> {
  const response = await fetch(`${apiBaseUrl}${path}`);

  if (!response.ok) {
    throw new Error(`Request failed: ${response.status}`);
  }

  return response.json() as Promise<T>;
}

export function fetchKnowledgeDocuments(): Promise<KnowledgeDocumentsResponse> {
  return requestJson<KnowledgeDocumentsResponse>("/knowledge/documents");
}

export function searchKnowledge(query: string, limit = 5): Promise<KnowledgeSearchResponse> {
  const params = new URLSearchParams({ q: query, limit: String(limit) });
  return requestJson<KnowledgeSearchResponse>(`/knowledge/search?${params.toString()}`);
}
